src/tarray.jl

```diff
@@ -37,15 +42,52 @@ TArray{T,N}(::UndefInitializer, d::Vararg{<:Integer,N}) where {T,N} = TArray{T,N
 TArray{T,N}(dim::NTuple{N,Int}) where {T,N} = TArray(T, dim)
 
 function TArray(T::Type, dim)
-    res = TArray{T,length(dim)}();
+    res = TArray{T, length(dim)}();
     n = n_copies()
     d = Array{T}(undef, dim)
-    task_local_storage(res.ref, (n,d))
+    _local_storage(res, (n, d))
     res
 end
 
 TArray(x::AbstractArray) = convert(TArray, x)
 
+# TArray House-Keeper
+
+const TArrayKeeper = Vector{WeakRef}()
+keep(x::TArray) = push!(TArrayKeeper, WeakRef(x))
+function copy_tarrays(task1::Task, task2::Task)
+    deleteat!(TArrayKeeper, map(x -> x == nothing, TArrayKeeper))
```

Comment:
A bit more efficient
```suggestion
    filter!(x -> x !== nothing, TArrayKeeper)
```
(but probably still not cheap)

Reply:
Done, BTW, it should be `!=` not `!==`.

> (but probably still not cheap)
This function only executes on task copying, so I think it's fine.

Reply:
Do you get incorrect results with `!==`? Usually it is better to compare singletons with `===` and `!==`, see e.g. the discussion in https://github.com/JuliaLang/julia/issues/35585#issuecomment-619053871.

Reply:
Yeah, `x` here is a `WeakRef`, `==` and `!=` are used to check its content, that what I want here:

```
julia> struct T i end

julia> t = T(1)
T(1)

julia> r = WeakRef(t)
WeakRef(T(1))

julia> r == nothing, r === nothing
(false, false)

julia> t = 0; GC.gc();

julia> r == nothing, r === nothing
(true, false)

julia> r != nothing, r !== nothing
(false, true)
```

Reply:
I see, it only works when one checks the field of the `WeakRef`:
```julia
julia> r.value === nothing
true
```

Reply:
It seems in the last case everything just compiles away but I could imagine it still being slightly more performant in general:
```julia
julia> f1(x) = x == nothing
f1 (generic function with 1 method)

julia> f2(x) = x.value === nothing
f2 (generic function with 1 method)

julia> r = WeakRef(T(1))
WeakRef(T(1))

julia> @btime f1($r);
  16.485 ns (0 allocations: 0 bytes)

julia> @btime f2($r);
  1.449 ns (0 allocations: 0 bytes)
```